src/precice/exceptions.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <stdexcept>
+
+namespace precice {
+
+class Error : public std::runtime_error {
+public:
+  Error(const std::string &what_arg)
+      : std::runtime_error(what_arg){};
+};
```

Comment:
I agree with the suggestion of @davidscn to keep the number of errors small for the beginning and find out what we need later. The main motivation here is to simplify using the correct error in the correct situation. Two concrete ideas:

* Only use a single exception for now: `Error`. We can extend this later? Then there is no risk of wrong or inconsistent usage of error classes, because everything is simple.
* The errors below are basically one error class per package. From my perspective this means that there is implicitly the rule: "Use ActionError, if you want to throw an error from `precice::action`". This rule is simple enough to not mess up things from the beginning, but we should also put this into the documentation of each error class.